File: app/src/main/java/com/example/myapplication/ui/memo/MemoFragment.java
package com.example.myapplication.ui.memo;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;

import com.example.myapplication.R;
import com.example.myapplication.RegisterActivity;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import androidx.lifecycle.ViewModelProviders;

public class MemoFragment extends Fragment implements View.OnClickListener {

    private MemoViewModel memoViewModel;

    public View onCreateView(@NonNull LayoutInflater inflater,
            ViewGroup container, Bundle savedInstanceState) {

        memoViewModel =
                ViewModelProviders.of(this).get(MemoViewModel.class);


        View root = inflater.inflate(R.layout.fragment_memo, container, false);
        final TextView textView = root.findViewById(R.id.text_home);


        memoViewModel.getText().observe(this, new Observer<String>() {
            @Override
            public void onChanged(@Nullable String s) {
                textView.setText(s);
            }
        });

        FloatingActionButton fab = (FloatingActionButton) root.findViewById(R.id.memo_fab);
        fab.setOnClickListener(this);
        return root;
    }

    @Override
    public void onClick(View view) {

        int id = view.getId();
        switch (id) {

            case R.id.memo_fab: {
                Intent intent = new Intent(getContext(), RegisterActivity.class);
                intent.putExtra("REGISTER_MODE", "MEMO");
                startActivity(intent);
                break;
            }
        }
    }
}

File: app/src/main/java/com/example/myapplication/ui/main/memo/MemoFragment.kt
package com.example.myapplication.ui.main.memo

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.PopupMenu
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer

import com.example.myapplication.R
import com.example.myapplication.database.AppDatabase
import com.google.android.material.floatingactionbutton.FloatingActionButton

import androidx.lifecycle.ViewModelProviders
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.myapplication.ui.register.memo.RegisterMemoActivity
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class MemoFragment : Fragment(), View.OnClickListener {

    private val db = AppDatabase.getInstance(context)
    private var memoViewModel: MemoViewModel? = null
    private var recyclerView: RecyclerView? = null
    private var memoAdapter: MemoAdapter? = null

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View? {


        val root = inflater.inflate(R.layout.fragment_memo, container, false)
        val layoutManager = LinearLayoutManager(context)

        recyclerView = root.findViewById<View>(R.id.memo_recycler) as RecyclerView
        recyclerView!!.layoutManager = layoutManager

        memoViewModel = ViewModelProviders.of(this).get(MemoViewModel::class.java)
        memoViewModel!!.context = context;
        memoViewModel!!.getData(db.memoDao()).observe(this, Observer { memos ->
            memoAdapter = MemoAdapter(memos, context)
            memoAdapter!!.setOnItemClickListener(memoViewModel!!.getItemClickListener())
            recyclerView!!.adapter = memoAdapter
        })

        val fab = root.findViewById<View>(R.id.memo_fab) as FloatingActionButton
        fab.setOnClickListener(this)
        return root
    }

    override fun onClick(view: View) {

        val id = view.id
        when (id) {

            R.id.memo_fab -> {
                val intent = Intent(context, RegisterMemoActivity::class.java)
                intent.putExtra("REGISTER_MODE", "MEMO")
                startActivity(intent)
            }
        }
    }




}